src/Microsoft.AspNet.TestHost/ClientHandler.cs

```diff
@@ -94,7 +94,7 @@ public ClientHandler(PathString pathBase, IHttpApplication<Context> application)
                     }
                 });
 
-            return await state.ResponseTask;
+            return await state.ResponseTask.ConfigureAwait(false);
```

Comment:
Though either the test change or the `ConfigureAwait` here will resolve the deadlock, it is suggested to do both here: http://blog.stephencleary.com/2012/07/dont-block-on-async-code.html